### PR TITLE
Feature/middleend

### PR DESCRIPTION
**Pull request by untra**
some preliminary work for consistent survey creation
- addresses #44 
- fixed typo in `src/handlers/survey_handler.py`
- added `serve-nogulp` to the makefile for start the webserver without running gulp
- changes to the `schema/page.json` to reflect what the actual implementation will look like
- :cocktail:

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -10,6 +10,9 @@ TESTS = $(wildcard test/*.py)
 serve:
 	export PROJECTPATH=${PROJECTPATH} && gulp prod-js && cd ./src && python3 server.py
```
Comment (antsankov) [inline Makefile:11]: Just want to emphasize why gulp is here. Right now, we don't share a common version of all.js which the HTML template points to. This makes sure that the react code is ready to go whenever the server is started.

File: package.json [hunk 2]
```diff
@@ -17,6 +17,7 @@
     "gulp-babel": "^5.3.0",
     "gulp-cache": "^0.3.0",
     "gulp-concat": "^2.6.0",
+    "gulp-expect-file": "^0.0.7",
```
Comment (antsankov) [inline package.json]: What is the purpose of this inclusion? Whas it in the gulpfile?

**Comment by antsankov** [comment]
Looks good. +1 to merge :egg:

File: Makefile (hunk 1 above)
Comment (untra) [inline Makefile:11]: I feel ya :+1: . I'm just working on some backend stuff, and gulp is failing silently like we mentioned on thursday. I don't need gulp or all.js at the moment, and consistent gulp compilation is a problem for the frontend to hash out (have fun with that, lol)

File: package.json (hunk 2 above)
Comment (untra) [inline package.json]: I was trying out [this](http://stackoverflow.com/a/22433775/1435958) solution. Didn't seem to work. I forgot to remove it. I'll remove it from the `package.json` and merge.

**Pull request merged**